NEWS.md

```diff
@@ -4,8 +4,8 @@ Julia v1.12 Release Notes
 New language features
 ---------------------
 
-- New option `--trim` for building "trimmed" binaries, where code not provably reachable from entry points
-  is removed. Entry points can be marked using `Base.Experimental.entrypoint` ([#55047]).
+- New option `--trim` creates smaller binaries by removing code that was proven not to be statically reachable from
```

Comment:
We still need to use "not proven" over "proven not" —

```suggestion
- New option `--trim` creates smaller binaries by removing code that was not proven to be statically reachable from
```

Reply:
Correct. I also think "statically" is kind of jargony and not necessary.

Reply:
done